src/proto/mod.rs

```diff
@@ -90,6 +90,18 @@ impl DerivedKey {
         )
     }
 
+    #[cfg(feature = "msextensions")]
+    pub fn from_client_key(k: Vec<u8>) -> Result<Self, KrbError> {
+        let k: [u8; AES_256_KEY_LEN] = k
+            .try_into()
+            .map_err(|_| KrbError::UnsupportedDerivedKeySize)?;
+        Ok(DerivedKey::Aes256CtsHmacSha196 {
+            k,
+            i: 0,
+            s: "".to_string(),
```

Comment:
@Firstyear I'd like your opinion here, since I'm not sure the appropriate way to handle this. See the way libhimmelblau currently decrypts the enc_part here: https://gitlab.com/samba-team/libhimmelblau/-/blob/main/src/auth.rs?ref_type=heads#L808

Reply:
So there are a few problems. MS still supports Aes128 (libkrimes does not). And then I wasn't sure what to do with these iteration and salt values. FYI, this `client_key` is encrypted using the session_key, and therefore TPM bound.

Reply:
The question is what the client key represents - realistically what you have is more likely the SessionKey ( https://github.com/kanidm/libkrimes/blob/main/src/proto/mod.rs#L276 ) since the DerivedKey generally comes from a password.

Within the operation of KRB, it's the SessionKey that's used in TGS REQ ( https://github.com/kanidm/libkrimes/blob/main/src/proto/request.rs#L232 ) and the ccache is storing the same https://github.com/kanidm/libkrimes/blob/main/src/ccache.rs#L217

So I suspect you actually need to drop this fn and use the SessionKey instead here :) 

The DerivedKey is only used during AS-REQ/REP, and I think from what you have indicated, you aren't doing an AS-REQ, but you just get an encrypted AS-REP that is bound to the TPM (Rather than the DerivedKey of the user). 


Re the AES128, we can add that later, it shouldn't be too hard after Aes256 at this point.

Reply:
It's not the session key. The client key is encrypted using the session key, and is included with the as-rep in the json response from Azure.
The source I referenced which I'm currently using is actually functional as is (I can use that client key to decrypt the enc_part of the asrep now). From what I can tell, it's used in exactly the same way as the DerivedKey, but there is no password, no salt, no iterations. I realize it 'generally comes from a password'. I'm saying in this case it does not. MS has done something weird here.

Reply:
Oh, we're talking about 2 different 'session keys'. Sorry, I don't really know.